specs/WinRT/WinRTAPIContracts.md

```diff
@@ -22,18 +22,22 @@ WinRT APIs.
    1. One contract per feature/component/sub-system
    2. Contracts will not span across a transport package/repo
 2. **Contract Version**<BR>
-Contracts versions are a single number with no minor version, no "x.y".  The contract version is
-frozen on a stable release.  Any changes after a stable release requires the version to be incremented.
+Contracts versions are a single number with no minor version, no "x.y". The contract version is
+frozen on a stable release. Any changes after a stable release requires the version to be incremented.
 3. **Enforcement**<BR>
 The build system has gates that enforce contract versioning to verify that contract is present,
  up to date, and there are no conflicts between APIs.
 4. **Contract Migration**<BR>
```

Comment:
I'd like to update the contracts to whatever new plan comes out of this spec. Since IsApiContractPresent isn't yet supported, this isn't a breaking change.

Reply:
> I'd like to update the contracts

Which contracts? Existing contracts defined in 1.0? That would be a breaking change.

If `IsApiContractPresent` is broken _today_ and fixed tomorrow, callers would expect it to work...but if they run into 1.0 artifacts it would produce different answers than probing 1.1+.

We can discuss further offline if you like

Reply:
Just keep in mind that renaming contracts will impact our ability to detect interface breaking changes using the winrt comparison tooling. If they're under different contracts, the tooling will likely treat them as unrelated APIs and not flag changes that it should.

Reply:
The comparison tooling compares namespace+typename, and then all the attributes on the typename. Contract is an attribute on the typename. Moving a type across namespaces is detected as "unexpected attribute change; ContractAttribute" ... but it _is_ a breaking change to rename a contract and move a type across contracts, as `IsApiContractPresent` on the old name will fail, even though the resolved-by-namespace-and-name type is present.

Reply:
Renaming or rearranging contracts isn't a breaking change in WinAppSDK yet because IsApiContractPresent isn't supported for non-OS types.

Reply:
Sorry, repeating an earlier comment, I can't find the page that says not to use contracts on non-OS types, and I must have had a typo when I tested it earlier.